### PR TITLE
Add add,remove, storage features to todo list app

### PR DESCRIPTION
**Pull request by kimothokamau**
This pull request adds the following features to the existing to-do list app:
- Adding new todo items
- Removing todo items
- Local storage perisistence

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @kimothokamau,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._
- [ ] The clear all completed functionality hasn't been added.
- [ ] When a task is marked as completed it doesn't get updated in localStorage.

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/frontend.js [hunk 1]
```diff
@@ -20,22 +27,30 @@ const displayTodos = (todos) => {
...
-    todoText.className = 'todo-text';
+    todoText.id = 'todo-text';
+    todoText.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter') {
+        createTodos(todoText.value);
+        ul.appendChild(todoItem(todos[todos.length - 1]));
 
-    newTodoCont.appendChild(todoText);
+        const clear = document.getElementById('clear');
+        ul.appendChild(clear);
```
Comment (mikenath223) [inline src/frontend.js:47]: - [x] Kindly add the `Clear all completed` functionality, which clears all completed todo items when clicked using the `Array.filter` method. Check [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20the%20%22Clear%20all%20completed%22%20button%20(use%20filter()%20method).).

File: src/frontend.js [hunk 2]
```diff
@@ -51,10 +66,13 @@ const displayTodos = (todos) => {
     checkbox.type = 'checkbox';
     checkbox.classList.add('completed');
     checkbox.name = 'completed';
+    checkbox.addEventListener('click', () => removeTodos(parseInt(todoLi.getAttribute('todo'), 10), checkbox.checked));
```
Comment (mikenath223) [inline src/frontend.js]: - [x] When a task is marked as completed, it should be updated in localStorage so refreshing the page doesn't remove the completed check. Check [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=All%20changes%20to%20the%20To%20Do%20List%20should%20be%20saved%20in%20local%20storage.).

File: src/frontend.js (hunk 1 above)
Comment (bellom) [reply src/frontend.js:47]: @kimothokamau, we still need to attend to this comment.

kindly add a function to clear all completed tasks using `Array.filter()` as stated in the project requirement.
Note: Make use of `Array.filer()` method.

Project Requirement Page:
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @kimothokamau 👏 

**Great work making `almost` all the necessary corrections from the previous review.** 👍

 ## Required Changes ♻️
_Check the comments above this review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/frontend.js (hunk 1 above)
Comment (kimothokamau) [reply src/frontend.js:47]: This has been resolved. Please check on this loom video that the 'Clear all completed' function is working as per the requirement. https://www.loom.com/share/85cad2d663ec474c94df14b19340b20d

**Comment by kimothokamau** [comment]
Hi @bellom 
I have confirmed that the 'Clear all completed' functionality is working as per the requirement. Kindly have a look at it again, I have recorded a video on how it is working from my side. See video in this link https://www.loom.com/share/85cad2d663ec474c94df14b19340b20d

**Comment by kimothokamau** [comment]
The loom video above and this live link https://kimothokamau.github.io/todo_list show that the 'clear all completed' functionality is working as per the requirement.

**Review by hemantso — APPROVED** [review]
Hi @kimothokamau, 

Well done on making the requested changes. 🚀 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers,
Hemant Soni
Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kimothokamau** [comment]
Thanks so much @hemant-soni-vst-au4 :)

File: src/frontend.js (hunk 1 above)
Comment (bellom) [reply src/frontend.js:47]: @kimothokamau, apologies 🙏  for the late response.

The files changes mislead me a little bit, I later saw you have the function and also you added `filter` method.

**Pull request merged**